docs/pages/kubernetes-access/discovery/aws.mdx

```diff
@@ -0,0 +1,232 @@
+---
+title: Teleport EKS Auto-Discovery (Preview)
+description: Auto-Discovery of EKS clusters in AWS cloud.
+---
+
+Similarly to [AKS Auto-Discovery](./azure.mdx), EKS Auto-Discovery can automatically 
+discover any EKS cluster and enroll it in Teleport if its tags match the 
+configured labels. 
+
+(!docs/pages/kubernetes-access/discovery/includes/step-description.mdx!)
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- An AWS account with permissions to create and attach IAM policies.
+- An AWS account with `system:masters` RBAC access to EKS clusters.
+- A host to run the Teleport Discovery and Kubernetes services.
+- One or more EKS clusters running. 
+
+## Known limitations
+
+Due to the authorization method that AWS has chosen for EKS clusters, it is not 
+possible for the Teleport process to configure the necessary permissions to forward 
+requests to EKS clusters.
+This limitation exists because the authorization method on EKS clusters requires that the IAM role -
+used for authentication - exists in the 
+[`aws-auth`](https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html) `ConfigMap`. 
+
+In this `ConfigMap`, IAM roles are mapped to Kubernetes RBAC users or groups that are then 
+used by Kubernetes RBAC to grant access permissions. If the mapping does not exist, it results in 
+unauthorized requests. Therefore, Teleport cannot edit the `ConfigMap`
+without first having access to the cluster.
+
+If Teleport is not running with the same IAM identity that creates all clusters, 
+please ensure you configure the required permissions as described in Step 2. 
+
+## Step 1/3. Set up AWS IAM credentials
+
+For Teleport to discover EKS clusters, the instance
+running the Discovery Service requires an IAM policy that allows Teleport to list and 
+describe EKS clusters:
+
+```js
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "eks:DescribeCluster",
+                "eks:ListClusters"
+            ],
+            "Resource": "*"
+        }
+    ]
+}      
+```
+
+## Step 2/3. Configure EKS cluster authorization
+
+
+When the Kubernetes Service uses an IAM role that is different from the one that 
+creates the clusters, you need to edit the `aws-auth` `Configmap` and append the 
+mapping between the Teleport IAM role and the RBAC permissions on each of the 
+discovered clusters. 
+
+For simplicity, it is possible to map the Teleport IAM role onto an existing 
+Kubernetes RBAC group such as `system:masters`, but Teleport recommends setting 
+up a separate Kubernetes RBAC group with the minimum required permissions.
+
+<Tabs>
+  <TabItem label="Minimum RBAC permissions (recommended)" >
```

Comment:
We should add a note somewhere in this section on how we recommend users automate this. Maybe say that if they have Terraform/Cloudformation that provisions EKS clusters and configures aws-auth, they can update that to include Teleport-required permissions? Otherwise users reading this might be wondering why automatic discovery requires manual steps for each cluster.